test/teststl.cpp

```diff
@@ -5200,6 +5200,16 @@ class TestStl : public TestFixture {
               "}\n",
               true);
         ASSERT_EQUALS("", errout.str());
+
+        check("bool g(int);\n"
+              "int f(const std::vector<int>& v) {\n"
+              "    int ret = 0;\n"
+              "    for (const auto i : v)\n"
+              "        if (!g(i))\n"
+              "            ret = 1;\n"
+              "    return ret;\n"
+              "}\n");
+        ASSERT_EQUALS("", errout.str());
```

Comment:
This should suggest `std::any_of, std::all_of, or std::none_of`.

Reply:
I don't think so, since those algorithms short-circuit. Presumably `g()` needs to be called for all elements.

Reply:
Then it should probably be `std::any_of, std::all_of, std::none_of, or std::accumulate`. Then you can use `std::accumulate` when `g()` has side effects.

Reply:
We could use `isConstExpression` to detect if the condition has side effects and then only suggest when there is no side effects.